affvisionpy_sample.py

```
@@ -1,11 +1,21 @@
 # !/usr/bin/env python3
+
 import argparse
 import sys
 import os
 import time
 import csv
 from collections import defaultdict
 
+# Before we import affvisionpy, check to see if LD_LIBRARY_PATH is set to a value that looks appropriate
+# If it's not set, importing affvisionpy will fail, and the error isn't that helpful.
+ld_library_path = os.environ.get("LD_LIBRARY_PATH")
+if (not ld_library_path or not os.path.isfile(ld_library_path + "/libaffectiva-vision.so")):
```

Can we on the fly try to see if we can locate the affvisionpy module, if we can we could generate the LD_LIBRARY_PATH similar to the data dir else we should throw an error

---

it's too late to set LD_LIBRARY_PATH inside the script, it needs to be set in advance.

---

I would maybe for convenience sake, set the LD_LIBRARY_PATH before the import of the affvisionpy module and check if the libraries are present at the path we expect them to be so that the users don't have to export them....we can meet to resolve this

---

it does no good to set LD_LIBRARY_PATH inside the script, it needs to be done externally, before the script is run.